### ISSUE TITLE
Fullscreen support

### ISSUE THREAD

**Issue by Valloric** [jwilm/alacritty#38]
Config should support something like `fullscreen: true` so that alacritty starts fullscreen/maximized.

**Comment by jwilm** [comment]
Sounds very reasonable. Should this big a config switch or a CLI switch?

**Comment by Valloric** [comment]
I'd prefer config.

**Comment by samdphillips** [comment]
Why not both?  CLI overrides config.

**Comment by Valloric** [comment]
![](http://thingiverse-production-new.s3.amazonaws.com/renders/d2/61/56/33/6b/Why_Not_Zoidberg_preview_featured.jpg)


I support both. :D

**Comment by akeif** [comment]
I'd love to get that also.
Config file would be more than enough.

**Comment by jez** [comment]
This could also be implemented as a keyboard shortcut (similar to the `Quit` action). I keep finding myself pressing <kbd>⌘</kbd><kbd>Enter</kbd> expecting the screen to maximize (old iTerm2 habits die hard 😛).

**Comment by jedahan** [comment]
Would also like to mention there are two kinds of fullscreen with macOS

Native which puts the window in its own 'space' and non-native which doesn't switch desktops. I prefer 'non-native', as seen in the latest update of mpv: https://github.com/mpv-player/mpv/pull/4676/files

**Comment by jayniz** [comment]
For me, the non-native fullscreen would be more important, as I never use the native full screen of MacOS (where you have to wait for animations to switch from one space to another). I'm with @jez and would love <kbd>⌘</kbd><kbd>Enter</kbd>

**Comment by jwilm** [comment]
Things have changed a lot in winit/Glutin since this was filed, and it may be possible to implement this now. Specifically, [`winit::Window::set_fullscreen`](https://docs.rs/winit/0.10.0/winit/struct.Window.html#method.set_fullscreen) is now available.

**Comment by djwbrown** [comment]
I tried adding the fullscreen startup (by config flag) feature here:
https://github.com/djwbrown/alacritty/tree/startup-fullscreen

So far it's not quite working on macOS. I just get a standard size window moved to the lower-left corner with broken radio-buttons in the menu bar. Only a green `+` remains for maximize, but typical close/minimize/native-fullscreen buttons are gone.

**Comment by felippemr** [comment]
@djwbrown Are you moving forward with your implementation? I'd like to work on this if you don't.

**Comment by felippemr** [comment]
@chrisduerr do we still want to do this?

**Comment by djwbrown** [comment]
@felippemr Feel free to take over and run with this! I haven't had time to continue working on it.

**Comment by chrisduerr** [comment]
@felippemr I don't see why not! :)

**Comment by chrisduerr** [comment]
Of course! If you have any questions feel free to ask in irc/github. :)

**Comment by metamaker** [comment]
@felippemr First pusher usually wins :)

**Comment by felippemr** [comment]
Hey, I've been out for a while but I'm planning to get back on this next week! 🎉 
Can you provide me some guidance on where I should look at to start working on this @chrisduerr ?

**Comment by chrisduerr** [comment]
Since this is related to configuration and window, the best places to start looking are probably the [`src/config.rs`](https://github.com/jwilm/alacritty/blob/master/src/config.rs) and [`src/window.rs`](https://github.com/jwilm/alacritty/blob/master/src/window.rs#L219) files.

If you have any more questions, just let me know. :)

**Comment by martskins** [comment]
I have some minimum viable version of this working right now (it in macOS it works only with native fullscreen). I tested it in linux as well and it seems to work fine.  It supports both the keybinding and config.  If anyone wants to review it, here it is: #1651

**Comment by hungpham3112** [comment]
Alacritty full screen mode doesn't apply opacity in Window 11.

**Comment by pabloqpacin** [comment]
Neither does it on Fedora

**Comment by TheArctesian** [comment]
For those using alacritty's now toml config change/or add this line in your `alacritty.toml` file. 

```py
startup_mode = "Windowed" | "Maximized" | "Fullscreen" | "SimpleFullscreen"
```

I just removed it from mine, here is my config for an example. It is not fancy but gets the job done. 

```toml
[colors.primary]
background = "#181818"

[window]
decorations = "None"
opacity = 0.7
padding = { x = 16 , y = 16 }

[env]
TERM = "alacritty"

[bell]
color = "#003300"
duration = 0

[font]
size = 13
offset = { x = 2, y = 1 }

[selection]
save_to_clipboard = true
```

**Comment by VladMorozDM** [comment]
> For those using alacritty's now toml config change/or add this line in your `alacritty.toml` file.
> 
> ```python
> startup_mode = "Windowed" | "Maximized" | "Fullscreen" | "SimpleFullscreen"
> ```
> 
> I just removed it from mine, here is my config for an example. It is not fancy but gets the job done.
> 
> ```toml
> [colors.primary]
> background = "#181818"
> 
> [window]
> decorations = "None"
> opacity = 0.7
> padding = { x = 16 , y = 16 }
> 
> [env]
> TERM = "alacritty"
> 
> [bell]
> color = "#003300"
> duration = 0
> 
> [font]
> size = 13
> offset = { x = 2, y = 1 }
> 
> [selection]
> save_to_clipboard = true
> ```
For completely new alacritters, don't forget to add "[window]" above
``python
[window]
startup_mode = "Windowed" | "Maximized" | "Fullscreen" | "SimpleFullscreen"
 ```

**Comment by hackermandh** [comment]
TOML Protip: Don't forget to either set dimensions to 0, 0, or just remove the option (which defaults to 0, 0 as of at least Alacritty 0.15.0):

```toml
[window]
dimensions = { columns = 0, lines = 0 }
startup_mode = "Maximized"
```

If you have the dimensions to anything other than 0, 0, it will override the "Maximized" option.
